internal/service/cloud/image.go

```diff
@@ -70,8 +71,20 @@ func (s *Service) lookupImageID(ctx context.Context, ms *scope.Machine) (string,
 		images = filterImagesByName(images, version)
 	}
 
-	if n := len(images); n != 1 {
-		return "", imageMatchError{imageIDs: getImageIDs(images), selector: imageSpec.Selector}
+	switch imageSpec.Selector.ResolutionPolicy {
+	case infrav1.ResolutionPolicyExact:
+		if len(images) != 1 {
+			return "", imageMatchError{imageIDs: getImageIDs(images), selector: imageSpec.Selector}
+		}
+	case infrav1.ResolutionPolicyNewest:
+		slices.SortFunc(images, func(lhs, rhs *sdk.Image) int {
+			// swap lhs and rhs to produce reverse order
+			return rhs.Metadata.CreatedDate.Compare(lhs.Metadata.CreatedDate.Time)
+		})
+	}
+
+	if len(images) == 0 {
```

Comment:
we can swipe this before the switch,

no need to process further if there are no images returned.